marlowe/src/Language/Marlowe/Client.hs

```diff
@@ -0,0 +1,250 @@
+-}
+commit :: (
+    MonadError WalletAPIError m,
+    WalletAPI m)
+    => (TxOut, TxOutRef)
+    -> ValidatorScript
+    -> [OracleValue Int]
+    -> [Choice]
+    -> IdentCC
+    -> Int
+    -> State
+    -> Contract
+    -> m ()
+commit txOut validator oracles choices identCC value inputState inputContract = do
+    bh <- slot
+    sig <- signature <$> myKeyPair
+    let inputCommand = Commit identCC sig
+    let input = Input inputCommand oracles choices
+    let scriptInValue@(Ledger.Value contractValue) = txOutValue . fst $ txOut
+    let scriptOutValue = Ledger.Value $ contractValue + value
+    let (expectedState, expectedCont, isValid) =
+            evalContract (PubKey 1) input bh scriptInValue scriptOutValue inputState inputContract
+    when (not isValid) $ throwOtherError "Invalid commit"
+    commit' txOut validator oracles choices identCC value expectedState expectedCont
+
+
+{-| Create a Marlowe Payment input transaction
+-}
+receivePayment :: (
+    MonadError WalletAPIError m,
+    WalletAPI m)
+    => (TxOut, TxOutRef)
+    -> ValidatorScript
+    -> [OracleValue Int]
+    -> [Choice]
+    -> IdentPay
+    -> Int
+    -> State
+    -> Contract
+    -> m ()
+receivePayment txOut validator oracles choices identPay value expectedState expectedCont = do
+    _ <- if value <= 0 then throwOtherError "Must commit a positive value" else pure ()
+    sig <- signature <$> myKeyPair
+    slot <- slot
+    let redeemer = createRedeemer (Payment identPay sig) oracles choices expectedState expectedCont
+    marloweTx redeemer txOut validator $ \ contractTxIn getTxOut contractValue -> do
+        let out = getTxOut (contractValue - value)
+        oo <- ownPubKeyTxOut (Ledger.Value value)
+        void $ createTxAndSubmit (intervalFrom slot) (Set.singleton contractTxIn) [out, oo]
+
+
+{-| Create a Marlowe Redeem input transaction
+-}
+redeem :: (
+    MonadError WalletAPIError m,
+    WalletAPI m)
+    => (TxOut, TxOutRef)
+    -> ValidatorScript
+    -> [OracleValue Int]
+    -> [Choice]
+    -> IdentCC
+    -> Int
+    -> State
+    -> Contract
+    -> m ()
+redeem txOut validator oracles choices identCC value expectedState expectedCont = do
+    _ <- if value <= 0 then throwOtherError "Must commit a positive value" else pure ()
+    sig <- signature <$> myKeyPair
+    slot <- slot
+    let redeemer = createRedeemer (Redeem identCC sig) oracles choices expectedState expectedCont
+    marloweTx redeemer txOut validator $ \ contractTxIn getTxOut contractValue -> do
+        let out = getTxOut (contractValue - value)
+        oo <- ownPubKeyTxOut (Ledger.Value value)
+        void $ createTxAndSubmit (intervalFrom slot) (Set.singleton contractTxIn) [out, oo]
+
+
+{-| Create a Marlowe SpendDeposit transaction
+
+    Spend the initial contract deposit payment.
+-}
+endContract :: (Monad m, WalletAPI m) => (TxOut, TxOutRef) -> ValidatorScript -> State -> m ()
```

Comment:
Why not call this `spendDeposit`? I do think that `endContract` makes a lot of sense, but I think being consistent with the Marlowe terminology should be most important here.